src/components/NotesMindMap/NotesMindMapReducer.js

```diff
@@ -14,114 +14,128 @@ import {
-      label: updatedNote.name,
-      name: updatedNote.name,
-      isNote: updatedNote.isNote,
-    });
-    return { ...state, notes };
+    let nodes = [...state.nodes];
+    nodes = nodes.filter(note => note.id !== updatedNote.id);
+    return {
+      ...state,
+      nodes: addNoteToGraph(nodes, updatedNote),
+    };
   };
 
   const handleCreateNoteSuccessAction = () => {
     const newNote = data;
-    let notes = [...state.notes];
+    let nodes = [...state.nodes];
     let edges = [...state.edges];
-    notes.push({
-      id: newNote.id,
-      name: newNote.name,
-      label: newNote.name,
-      isNote: newNote.isNote,
-      group: 'children',
-    });
     edges.push({ from: newNote.parent.id, to: newNote.id });
-    notes = addGroupTagToNote(notes, state.selectedNote.id);
-    return { ...state, notes, edges };
+    nodes = addGroupTagToNodes(addNoteToGraph(nodes, newNote), edges);
+    return {
+      ...state,
+      nodes,
+      edges,
+    };
   };
 
   const handleDeleteNoteRequestSuccessAction = () => {
     const noteToDelete = data;
-    let notes = [...state.notes];
-    let edges = [...state.edges];
-    let graph = removeNoteFromGraph(notes, edges, noteToDelete);
-    notes = graph.notes;
-    edges = graph.edges;
-    return { ...state, notes, edges };
+    const { nodes, edges } = removeNodeFromGraph(
+      state.nodes,
+      state.edges,
+      noteToDelete,
+    );
+    return {
+      ...state,
+      nodes,
+      edges,
+    };
   };
 
   const handleChangeParentNoteRequestSuccess = () => {
     const noteId = data.noteId;
     const newParentId = data.newParentId;
     let edges = [...state.edges];
-    let notes = [...state.notes];
+    let nodes = [...state.nodes];
     const oldParent = edges.find(edge => edge.to === noteId);
     edges = edges.filter(edge => edge.to !== noteId);
     edges.push({ from: newParentId, to: noteId });
 
-    notes = updateGroupOfOldParent(notes, edges, newParentId, oldParent.from);
+    nodes = updateGroupOfOldParent(nodes, edges, newParentId, oldParent.from);
     return {
       ...state,
       isChangeParentModeActive: false,
       showNoteNameEditor: false,
       edges,
-      notes,
+      nodes,
     };
   };
 
   const handleChangeSelectedNoteAction = () => {
-    const newState = { ...state };
-    newState.selectedNote = data.note;
-    return { ...state, selectedNote: data.note };
+    return {
+      ...state,
+      selectedNote: data.note,
+    };
   };
 
   const addRootToGraph = () => {
-    let notes = [...state.notes];
-    notes.push({ id: data.id, label: data.name });
-    return { ...state, notes };
+    let nodes = [...state.nodes];
+    nodes.push({ id: data.id, label: data.name, name: data.name });
+    return {
+      ...state,
+      nodes,
+    };
+  };
+
+  const addNoteToGraph = (nodes, newNote) => {
+    nodes.push({
+      id: newNote.id,
+      label: newNote.name,
+      name: newNote.name,
+      isNote: newNote.isNote,
+    });
```

Comment:
So basically it's `nodes.push({ ...newNote, label: newNote.name })`?

Reply:
Do we need to add other attributes of the fetched data i.e `mimeType, parents, parent`?
If yes then we can go for it.

Reply:
Oh didnt think about those :)
No we dont need them, so you are doing the right thing.

Reply:
Since the `newNote` has other attributes like `mimeType`, so it's not a note? But a folder?
So this function should be accepting `noteFolder` instead of `newNote`?
And so this function does 2 things:
- creates a node object from a folder object
- adds the created node to the graph

The function name `addNodeToGraph` makes me think that it simply adds a node that is passed to the graph. But turns out it accepts a folder,  creates a new object from it, and adds that to the graph.
So it's a bit confusing. Can we do something to make things more clear?

Right now here `newNote` has `mimeType` property, which is a bit confusing, since notes have no `mimeType`. Google Drive files have `mimeType`.

Reply:
Good point. It is confusing. We should rename it `newNode` instead, as we are basically adding a node to the graph. A node can be either a folder or a text file.